pkg/replicate/scheme.go

```diff
@@ -274,6 +298,23 @@ func (rs *replicationScheme) ensureBlockIsReplicated(ctx context.Context, id uli
 		return errors.Wrap(err, "upload meta file")
 	}
 
+	if rs.markAfter {
+		deletionMarkFile := path.Join(id.String(), metadata.DeletionMarkFilename)
```

Comment:
`MarkForDeletion` already checks for this, no? It could get quite expensive to check twice if one is replicating, let's say, 10000 blocks.

Reply:
I thought it would be very confusing to users, because `MarkForDeletion` will log with warn level following: `requested to mark for deletion, but file already exists; this should not happen; investigate`. Don't know if it is a good idea to edit `MarkForDeletion` method to implement optional warn bypass, or maybe this log message is fine.

Reply:
@GiedriusS can you please take a look?